src/lib.rs

```diff
@@ -264,7 +264,7 @@ impl<N: Name> TypeSchema<N> {
     }
     fn free_vars_internal(&self, ctx: &Context<N>, s: &mut HashSet<Variable>) {
```

Comment:
We should get rid of the `ctx` argument for `TypeSchema::free_vars`. Also, we should probably rename `bound_variables` to `bound_vars`

Reply:
Yup - just realized this myself reviewing the code this morning.